src/sdk.spec.js

```diff
@@ -125,17 +125,87 @@ describe('OpenForm', () => {
     expect(form.clientBaseUrl).toEqual('http://localhost/some-subpath');
   });
 
-  it('should correctly set the formUrl (hash fragment routing)', () => {
+  it("shouldn't take basepath into account (hash based routing)", () => {
     mswServer.use(...apiMocks);
-    window.history.pushState({}, 'Dummy title', '/some-server-side/path');
+    window.history.pushState({}, 'Dummy title', '/some-path');
     const formRoot = document.createElement('div');
     const form = new OpenForm(formRoot, {
       baseUrl: BASE_URL,
-      basePath: '/some-subpath/',
       formId: '81a22589-abce-4147-a2a3-62e9a56685aa',
       useHashRouting: true,
     });
 
-    expect(form.clientBaseUrl).toEqual('http://localhost/some-server-side/path#/some-subpath');
+    expect(form.clientBaseUrl).toEqual('http://localhost/');
```

Comment:
I think discarding `window.location.pathname` with hash based routing is wrong though, this should be included (but the variant is that we don't let `basePath` overwrite this value).

Imagine that a CMS has `https://example.com/forms/form1` where you start the form - the SDK will then initialize and point you to `https://example.com/forms/form1#/startpagina` and the submission will be created in the backend.

The `submission.formUrl` in this case needs to be `https://example.com/forms/form1` so that our backend can redirect you back to this (server-side) page and the SDK router can take over.

```suggestion
    expect(form.clientBaseUrl).toEqual('http://localhost/some-path');
```